scripts/firedrake-install

```diff
@@ -488,6 +488,12 @@ pipinstall = pip + ["install",
                     "--no-build-isolation",
                     "--no-binary", ",".join(wheel_blacklist)]
 
+if mode == "install":
+    use_petsc4py_in_petsc = "petsc4py" not in branches
+else:
+    src_petsc4py_exists = os.path.exists(os.path.join(firedrake_env, "src/petsc4py"))
+    use_petsc4py_in_petsc = not (src_petsc4py_exists and args.honour_petsc_dir)
```

Comment:
In args.honour_petsc_dir case, should we not check PETSC_DIR/PETSC_ARCH/src/bindings/python ?

Reply:
`use_petsc4py_in_petsc` should always be true, but complain if out of tree petsc doesn't contain bindings/python

Reply:
AKA `raise InstallError`

Reply:
Done.